kairos-server/src/routes/fetch.rs

```diff
@@ -0,0 +1,43 @@
+use axum::{extract::State, Json};
+use axum_extra::routing::TypedPath;
+use tracing::instrument;
+use serde::Deserialize;
+use chrono::NaiveDateTime;
+use kairos_data::transaction::*;
+
+use crate::{state::ServerState, AppErr};
+
+#[derive(Deserialize, Debug)]
+pub struct QueryTransactionsPayload {
+    sender: Option<String>,
+    min_timestamp: Option<String>,
+    max_timestamp: Option<String>,
+    min_amount: Option<i64>,
+    max_amount: Option<i64>,
+    recipient: Option<String>,
+}
+
+#[derive(TypedPath)]
+#[typed_path("/api/v1/query_transactions")]
+pub struct QueryTransactionsPath;
+
+#[instrument(level = "trace", skip(state), ret)]
+pub async fn query_transactions_handler(
+    _: QueryTransactionsPath,
+    State(state): State<ServerState>,
+    Json(payload): Json<QueryTransactionsPayload>,
+) -> Result<Json<Vec<Transaction>>, AppErr> {
+    let filter = TransactionFilter {
+        sender: payload.sender,
+        min_timestamp: payload.min_timestamp.and_then(|ts| NaiveDateTime::parse_from_str(&ts, "%Y-%m-%d %H:%M:%S").ok()),
+        max_timestamp: payload.max_timestamp.and_then(|ts| NaiveDateTime::parse_from_str(&ts, "%Y-%m-%d %H:%M:%S").ok()),
+        min_amount: payload.min_amount,
+        max_amount: payload.max_amount,
+        recipient: payload.recipient,
+    };
+
+    let transactions = get(state.pool.clone(), filter)
```

Comment:
If we make `get` take a reference we could avoid cloning here, or do we need to take ownership?

Reply:
Nah we don't need to clone, fixed